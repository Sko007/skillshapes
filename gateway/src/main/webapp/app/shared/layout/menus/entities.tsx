import React from 'react';
import MenuItem from 'app/shared/layout/menus/menu-item';
import { DropdownItem } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Translate, translate } from 'react-jhipster';
import { NavLink as Link } from 'react-router-dom';
import { NavDropdown } from './menu-components';

export const EntitiesMenu = props => (
  <NavDropdown
    icon="th-list"
    name={translate('global.menu.entities.main')}
    id="entity-menu"
    style={{ maxHeight: '80vh', overflow: 'auto' }}
  >
    <MenuItem icon="asterisk" to="/user-profile">
      <Translate contentKey="global.menu.entities.userProfile" />
    </MenuItem>
    <MenuItem icon="asterisk" to="/skill-shape">
      <Translate contentKey="global.menu.entities.skillShape" />
    </MenuItem>
    <MenuItem icon="asterisk" to="/profile-skill-value">
      <Translate contentKey="global.menu.entities.profileSkillValue" />
    </MenuItem>
    <MenuItem icon="asterisk" to="/skill">
      <Translate contentKey="global.menu.entities.skill" />
    </MenuItem>
    {/* jhipster-needle-add-entity-to-menu - JHipster will add entities to the menu here */}
  </NavDropdown>
);
