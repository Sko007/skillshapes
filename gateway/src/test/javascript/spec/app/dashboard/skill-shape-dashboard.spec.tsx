import React from 'react';
import { shallow } from 'enzyme';

import sinon from 'sinon';

import LoadingBar from 'react-redux-loading-bar';
import { Navbar, Nav } from 'reactstrap';
import { DashboardMenu } from 'app/shared/layout/menus';
import Header from 'app/shared/layout/header/header';

describe('Dashboard', () => {
  let mountedWrapper;

  const localeSpy = sinon.spy();

  const devProps = {
    isAuthenticated: true,
    isAdmin: true,
    currentLocale: 'en',
    onLocaleChange: localeSpy,
    ribbonEnv: 'dev',
    isInProduction: false,
    isSwaggerEnabled: true,
  };
  const prodProps = {
    ...devProps,
    ribbonEnv: 'prod',
    isInProduction: true,
    isSwaggerEnabled: false,
  };
  const userProps = {
    ...prodProps,
    isAdmin: false,
  };
  const guestProps = {
    ...prodProps,
    isAdmin: false,
    isAuthenticated: false,
  };

  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Header {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  // All tests will go here
  it('Renders a Header component with dashboard menu.', () => {
    const component = wrapper();
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    const nav = component.find(Nav);
    expect(nav.find(DashboardMenu).length).toEqual(1);
  });

  it('Renders a Header component in prod profile with logged in User', () => {
    const nav = wrapper(userProps).find(Nav);
    expect(nav.find(DashboardMenu).length).toEqual(1);
  });

  it('Renders no dashboard with no logged in User', () => {
    const nav = wrapper(guestProps).find(Nav);
    expect(nav.find(DashboardMenu).length).toEqual(0);
  });
});
