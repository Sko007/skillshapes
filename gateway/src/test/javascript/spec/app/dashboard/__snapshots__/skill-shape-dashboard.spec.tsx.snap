// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Renders a Header component with dashboard menu. 1`] = `
<div
  id="app-header"
>
  <div
    className="ribbon dev"
  >
    <a
      href=""
    >
      <Translate
        component="span"
        contentKey="global.ribbon.dev"
      />
    </a>
  </div>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <Navbar
    className="jh-navbar"
    dark={true}
    expand="sm"
    fixed="top"
    tag="nav"
  >
    <NavbarToggler
      aria-label="Menu"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Component />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        id="header-tabs"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <Component />
        <Component />
        <Component />
        <Component
          showSwagger={true}
        />
        <Component
          currentLocale="en"
          onClick={[Function]}
        />
        <Component
          isAuthenticated={true}
        />
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
